wfe2/wfe.go

```diff
@@ -490,6 +490,13 @@ func (wfe *WebFrontEndImpl) NewAccount(
 
 	existingAcct, err := wfe.SA.GetRegistrationByKey(ctx, key)
 	if err == nil {
+		if existingAcct.Status == core.StatusDeactivated {
```

Comment:
If its not too much trouble could you also update the `wfe2/wfe.go` function [`Account`](https://github.com/letsencrypt/boulder/blob/b24fe7c3ea47da5b810b1507d8950fc87f24dc9a/wfe2/wfe.go#L1123-L1124) to handle deactivated accounts the same way you do here? It would also need a short unit test.

Presently I think its possible for a deactivated account to poll its own account information and I think it makes sense to return an unauthorized problem in that case as well.

Reply:
Are you sure `Account` needs the check? It calls `validPOSTForAccount`, which calls `validJWSForAccount`, which calls `lookupJWK`, which checks for deactivation. The check is necessary here because NewAccount uses `validSelfAuthenticatedPOST`.

Reply:
Aw nuts, you're totally right @jsha. Thanks for catching that. Sorry @klebervirgilio - I sent you on a goose chase with this piece of feedback.